### ISSUE TITLE
Ability to set ultimate trust for other keys

### ISSUE THREAD

**Issue by jelmer** [open-keychain/open-keychain#1312]
It would be great it if it was possible to set ultimate trust for keys for which the private keys are not present on the phone. E.g. I don't have my main private key on my phone.

**Comment by dschuermann** [comment]
Out of scope for OpenKeychain.
Please generate a new key solely for this phone and use it to locally confirm other keys. You don't have to use it for actual communication just to work around this limitation.

**Comment by jelmer** [comment]
On Fri, May 29, 2015 at 05:48:07AM -0700, Dominik Schürmann wrote:
> Out of scope for OpenKeychain.
> Please generate a new key solely for this phone and use it to locally confirm other keys. You don't have to use it for actual communication just to work around this limitation.
I have a local key on my phone. The suggested workaround would require me to
locally (on the phone) re-sign all keys that I have signed with my main
(off-phone) key, just to be able to easily see which keys I trust.

**Comment by dschuermann** [comment]
> The suggested workaround would require me to locally (on the phone) re-sign all keys that I have signed with my main (off-phone) key, just to be able to easily see which keys I trust.

Which is the same amount of work compared to setting ultimate trust to every single key, or am I missing something?

**Comment by jelmer** [comment]
On Fri, May 29, 2015 at 06:36:43AM -0700, Dominik Schürmann wrote:
> > The suggested workaround would require me to locally (on the phone) re-sign all keys that I have signed with my main (off-phone) key, just to be able to easily see which keys I trust.
> 
> Which is the same amount of work compared to setting ultimate trust to every single key, or am I missing something?
Just setting ultimate trust for the main key should be sufficient.

At least that's the way it works with gnupg. I also use this on servers where I
read my mail but don't have my private key stored. Setting the trust for my
main key to "ultimate" there means I at least get GPG signature validation, including
whether I have signed the key or not.

Jelmer

**Comment by Valodim** [comment]
the canonical solution is to your key as one of your own, but without private key material, sort of how yubikey-bound keys work. this is not a ui option yet, but would be easy to implement

**Comment by jelmer** [comment]
Thanks for reopening. I probably should have phrased the bug report somewhat differently; I care about being able to mark a key as my own for which the private data is not on the phone. How that is done (ultimate trust or some other way) is probably an implementation detail.

**Comment by Valodim** [comment]
this is mostly a problem of where to position the option in the ui. do you have a spots in mind where this might fit, or where you maybe looked for it before reporting?

I suppose somewhere under extended information is fine, since it's quite the poweruser option.

**Comment by jelmer** [comment]
Extended information does seem like a reasonable place for it. That's where I first checked to see if there was such an option.

**Comment by Valodim** [comment]
sounds good. I planned to create an advanced key operations tab there anyways, I will place the operation there

**Comment by cassdelacruzmunoz** [comment]
Hey, are there any updates on this? It's been open for the last 3 years without any update. And with the most recent version in the play store, it certainly isn't resolved.

**Comment by Valodim** [comment]
nope, and it's not on our roadmap. feel free to give it a shot though, PRs welcome! basically there just needs to be some UI that triggers a PromoteKeyOperation on the key
